### PR TITLE
Add some pre-build funtions and tests

### PR DESCRIPTION
**Pull request by qtzhuang**
1. add AngleEmbedding
2. add BasicEntangleLayer
3. For supporting the above functions, add "add_gates" function in QuantumCircuit temporarily
4. tests or example has been also added
5. amplitude embedding will be added soon

### CODE REVIEW

File: quafu/algorithms/layers/basic_entangle.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+class BasicEntangleLayers:
+    def __init__(self, weights, num_qubits, rotation="X"):
+        """
+        Args:
+            weights(array): Each weight is used as a parameter for the rotation
+            num_qubits(int): the number of qubit
+            rotation(str): one-parameter single-qubit gate to use
+        """
+        weights = np.asarray(weights)
+        """convert weights to numpy array if weights is list otherwise keep unchanged"""
```
Comment (Zhaoyilunnn) [inline quafu/algorithms/layers/basic_entangle.py:31]: minor: inline comments should start with "#"

File: quafu/algorithms/layers/basic_entangle.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                gate = self.op(pos=i, paras=self.weights[layer][i])
+                gate_list.append(gate)
+                
+            # if num_qubits equals two, it just need to apply CNOT one time
+            if self.num_qubits == 2:
+                gate_list.append(qeg.CXGate(0,1))
+
+            elif self.num_qubits > 2:
+                for i in range(self.num_qubits):
+                    gate_list.append(qeg.CXGate(i,(i+1)%self.num_qubits))
```
Comment (Zhaoyilunnn) [inline quafu/algorithms/layers/basic_entangle.py:72]: missing space between `i` and `(i+1) ...`

**Review by Zhaoyilunnn — APPROVED** [review]
Thanks for the great work. In general this PR looks good to me, future improvements may include

1. using some tools like black to unity the coding style, this can easily avoid some minors like missing white spaces.
2. refer to pennylane or other opensource projects and read their unit tests, figuring out how they write ut for circuit templates and improve our ut.

File: quafu/algorithms/layers/basic_entangle.py (hunk 2 above)
Comment (qtzhuang) [reply quafu/algorithms/layers/basic_entangle.py:72]: thanks～，I humbly accept the above suggestions and make better work~

**Pull request merged**